homeassistant/components/skybell/light.py

```diff
@@ -1,90 +1,76 @@
-    return int((level * 100) / 255)
+    """Set up Skybell switch."""
+    async_add_entities(
+        SkybellLight(coordinator)
+        for coordinator in hass.data[DOMAIN][entry.entry_id].values()
+    )
 
 
-def _to_hass_level(level):
-    """Convert the given Skybell (0-100) light level to Home Assistant (0-255)."""
-    return int((level * 255) / 100)
+class SkybellLight(SkybellEntity, LightEntity):
+    """A light implementation for Skybell devices."""
 
+    _attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS, COLOR_MODE_HS}
+    coordinator: SkybellDataUpdateCoordinator
 
-class SkybellLight(SkybellDevice, LightEntity):
-    """A binary sensor implementation for Skybell devices."""
-
-    def __init__(self, device):
+    def __init__(
+        self,
+        coordinator: SkybellDataUpdateCoordinator,
+    ) -> None:
         """Initialize a light for a Skybell device."""
-        super().__init__(device)
-        self._name = self._device.name
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+        super().__init__(coordinator)
+        self._attr_name = coordinator.name
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{self.name}"
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on the light."""
         if ATTR_HS_COLOR in kwargs:
             rgb = color_util.color_hs_to_RGB(*kwargs[ATTR_HS_COLOR])
-            self._device.led_rgb = rgb
+            await self.coordinator.device.async_set_setting(ATTR_HS_COLOR, rgb)
         elif ATTR_BRIGHTNESS in kwargs:
-            self._device.led_intensity = _to_skybell_level(kwargs[ATTR_BRIGHTNESS])
+            level = int((kwargs[ATTR_BRIGHTNESS] * 100) / 255)
+            await self.coordinator.device.async_set_setting(ATTR_BRIGHTNESS, level)
         else:
-            self._device.led_intensity = _to_skybell_level(255)
+            await self.coordinator.device.async_set_setting(ATTR_BRIGHTNESS, 100)
 
-    def turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn off the light."""
-        self._device.led_intensity = 0
+        await self.coordinator.device.async_set_setting(ATTR_BRIGHTNESS, 0)
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if device is on."""
-        return self._device.led_intensity > 0
+        return self.coordinator.device.led_intensity > 0
 
     @property
-    def brightness(self):
+    def brightness(self) -> int:
         """Return the brightness of the light."""
-        return _to_hass_level(self._device.led_intensity)
+        return int((self.coordinator.device.led_intensity * 255) / 100)
 
     @property
-    def hs_color(self):
+    def hs_color(self) -> tuple[float, float]:
         """Return the color of the light."""
-        return color_util.color_RGB_to_hs(*self._device.led_rgb)
```

Comment:
Why is this using `hs_color` when the device appears to use native RGB?

Reply:
This was the old behavior before my work, fixed